posts.go

```diff
@@ -1193,6 +1193,16 @@ func getRawCollectionPost(app *App, slug, collAlias string) *RawPost {
 	}
 }
 
+func isRaw(r *http.Request) bool {
+	vars := mux.Vars(r)
+	slug := vars["slug"]
+
+	isJSON := strings.HasSuffix(slug, ".json")
+	isXML := strings.HasSuffix(slug, ".xml")
+	isMarkdown := strings.HasSuffix(slug, ".md")
+	return strings.HasSuffix(slug, ".txt") || isJSON || isXML || isMarkdown
+}
+
 func viewCollectionPost(app *App, w http.ResponseWriter, r *http.Request) error {
 	vars := mux.Vars(r)
```

Comment:
these 7 lines are redundant now with `isRaw` above.

Reply:
I know it's not the cleanest, but we need the result of `isRaw()` for `CollectionPostOrStatic()`, and we need all the vars in that logic below. I put them close together so we could have both without a ton of refactoring. Maybe for now I should add some comments mentioning that the two places need to be kept in parity?

Reply:
sure that's probably good

Reply:
done :+1: